commons/protocol/src/main/scala/sbt/protocol/Values.scala

```diff
@@ -66,14 +66,14 @@ final case class TaskSuccess(value: BuildValue) extends TaskResult {
   override def isSuccess = true
   override def resultWithCustomThrowable[A, B <: Throwable](implicit unpickleResult: Unpickler[A], unpickleFailure: Unpickler[B]): Try[A] = {
     value.value[A] match {
-      case Some(v) => Success(v)
-      case None => Failure(new Exception(s"Failed to deserialize ${value.serialized}"))
+      case Success(v) => Success(v)
+      case Failure(t) => Failure(new Exception(s"Failed to deserialize ${value.serialized}", t))
```

Comment:
the "outer" exception here probably adds nothing, relevant context should be in `t` already (may not be because sometimes pickling throws lame exceptions, but we should really fix them there). 
I'd be included to do `case f: Failure => f`